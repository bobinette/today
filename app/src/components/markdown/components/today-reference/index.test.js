import React from 'react';
import { mount, shallow } from 'enzyme';
import { fromJS } from 'immutable';

import * as matchers from 'jest-immutable-matchers';

import { TodayReference } from '.';

beforeEach(() => {
  jest.addMatchers(matchers);
});

test('render reference without auto load', () => {
  const props = {
    autoLoad: false,
    fetchLog: jest.fn(),
    uuid: '1',
  };

  const tree = shallow(<TodayReference {...props} />);
  expect(tree).toMatchSnapshot();
  expect(tree.state().loading).toBe(false);
  expect(props.fetchLog).not.toHaveBeenCalled();
});

test('render reference without auto load - loading', () => {
  const props = {
    autoLoad: false,
    fetchLog: jest.fn(),
    uuid: '1',
  };

  const tree = shallow(<TodayReference {...props} />);
  tree.setState({ loading: true });
  expect(tree).toMatchSnapshot();
  expect(props.fetchLog).not.toHaveBeenCalled();
});

test('render reference with auto load - log exists', async () => {
  const log = { content: 'pizza yolo' };
  const p = Promise.resolve({ log });
  const props = {
    autoLoad: true,
    fetchLog: jest.fn(() => p),
    uuid: '1',
  };

  const tree = shallow(<TodayReference {...props} />);
  await p;
  expect(props.fetchLog).toHaveBeenCalledWith('1');
  expect(tree.state().loading).toBe(false);
  expect(tree.state().log).toEqualImmutable(fromJS(log));
  // This is the secret to rerender the component after the
  // state has been updated
  tree.update();
  expect(tree).toMatchSnapshot();

  expect(tree.find('.TodayReference__Load').length).toBe(0);
  expect(tree.find('.TodayReference__Error').length).toBe(0);
  expect(tree.find('.TodayReference__Loading').length).toBe(0);
  expect(tree.find('Markdown').length).toBe(1);
});

test('render reference with auto load - errored', async () => {
  const error = { message: 'error' };
  const p = Promise.resolve({ error });
  const props = {
    autoLoad: true,
    fetchLog: jest.fn(() => p),
    uuid: '1',
  };

  const tree = shallow(<TodayReference {...props} />);
  await p;
  expect(props.fetchLog).toHaveBeenCalledWith('1');
  expect(tree.state().loading).toBe(false);
  expect(tree.state().error).toEqual(error);

  tree.update();
  expect(tree).toMatchSnapshot();

  expect(tree.find('.TodayReference__Load').length).toBe(0);
  expect(tree.find('.TodayReference__Error').length).toBe(1);
  expect(tree.find('.TodayReference__Loading').length).toBe(0);
  expect(tree.find('Markdown').length).toBe(0);
});
