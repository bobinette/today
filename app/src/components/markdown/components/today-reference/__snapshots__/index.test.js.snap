// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render reference with auto load - errored 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodayReference
    autoLoad={true}
    fetchLog={
      [MockFunction] {
        "calls": Array [
          Array [
            "1",
          ],
        ],
      }
    }
    uuid="1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Tooltip
        align={Object {}}
        arrowContent={null}
        destroyTooltipOnHide={false}
        mouseEnterDelay={0.3}
        mouseLeaveDelay={0.1}
        overlay="1"
        placement="bottom"
        prefixCls="rc-tooltip"
        trigger={
          Array [
            "hover",
          ]
        }
        visible={false}
      >
        <div
          className="TodayReference__Error flex flex-align-items-center"
        >
          <i
            className="fas fa-exclamation-circle TodayReference__Icon"
          />
          <span>
            Could not load 
            <code>
              1
            </code>
            :
            <br />
            error
          </span>
        </div>
      </Tooltip>,
      "className": "TodayReference TodayReference__Errored",
      "onBlur": [Function],
      "onFocus": [Function],
      "onMouseOut": [Function],
      "onMouseOver": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": Object {},
        "arrowContent": null,
        "children": <div
          className="TodayReference__Error flex flex-align-items-center"
        >
          <i
            className="fas fa-exclamation-circle TodayReference__Icon"
          />
          <span>
            Could not load 
            <code>
              1
            </code>
            :
            <br />
            error
          </span>
        </div>,
        "destroyTooltipOnHide": false,
        "mouseEnterDelay": 0.3,
        "mouseLeaveDelay": 0.1,
        "overlay": "1",
        "placement": "bottom",
        "prefixCls": "rc-tooltip",
        "trigger": Array [
          "hover",
        ],
        "visible": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="fas fa-exclamation-circle TodayReference__Icon"
            />,
            <span>
              Could not load 
              <code>
                1
              </code>
              :
              <br />
              error
            </span>,
          ],
          "className": "TodayReference__Error flex flex-align-items-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-exclamation-circle TodayReference__Icon",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Could not load ",
                <code>
                  1
                </code>,
                ":",
                <br />,
                "error",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Could not load ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1",
                },
                "ref": null,
                "rendered": "1",
                "type": "code",
              },
              ":",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "error",
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Tooltip
          align={Object {}}
          arrowContent={null}
          destroyTooltipOnHide={false}
          mouseEnterDelay={0.3}
          mouseLeaveDelay={0.1}
          overlay="1"
          placement="bottom"
          prefixCls="rc-tooltip"
          trigger={
            Array [
              "hover",
            ]
          }
          visible={false}
        >
          <div
            className="TodayReference__Error flex flex-align-items-center"
          >
            <i
              className="fas fa-exclamation-circle TodayReference__Icon"
            />
            <span>
              Could not load 
              <code>
                1
              </code>
              :
              <br />
              error
            </span>
          </div>
        </Tooltip>,
        "className": "TodayReference TodayReference__Errored",
        "onBlur": [Function],
        "onFocus": [Function],
        "onMouseOut": [Function],
        "onMouseOver": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": Object {},
          "arrowContent": null,
          "children": <div
            className="TodayReference__Error flex flex-align-items-center"
          >
            <i
              className="fas fa-exclamation-circle TodayReference__Icon"
            />
            <span>
              Could not load 
              <code>
                1
              </code>
              :
              <br />
              error
            </span>
          </div>,
          "destroyTooltipOnHide": false,
          "mouseEnterDelay": 0.3,
          "mouseLeaveDelay": 0.1,
          "overlay": "1",
          "placement": "bottom",
          "prefixCls": "rc-tooltip",
          "trigger": Array [
            "hover",
          ],
          "visible": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fas fa-exclamation-circle TodayReference__Icon"
              />,
              <span>
                Could not load 
                <code>
                  1
                </code>
                :
                <br />
                error
              </span>,
            ],
            "className": "TodayReference__Error flex flex-align-items-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-exclamation-circle TodayReference__Icon",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Could not load ",
                  <code>
                    1
                  </code>,
                  ":",
                  <br />,
                  "error",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Could not load ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1",
                  },
                  "ref": null,
                  "rendered": "1",
                  "type": "code",
                },
                ":",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                "error",
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`render reference with auto load - log exists 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodayReference
    autoLoad={true}
    fetchLog={
      [MockFunction] {
        "calls": Array [
          Array [
            "1",
          ],
        ],
      }
    }
    uuid="1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Tooltip
        align={Object {}}
        arrowContent={null}
        destroyTooltipOnHide={false}
        mouseEnterDelay={0.3}
        mouseLeaveDelay={0.1}
        overlay="1"
        placement="bottom"
        prefixCls="rc-tooltip"
        trigger={
          Array [
            "hover",
          ]
        }
        visible={false}
      >
        <Markdown
          autoLoadReferences={false}
          className=""
          text="pizza yolo"
        />
      </Tooltip>,
      "className": "TodayReference",
      "onBlur": [Function],
      "onFocus": [Function],
      "onMouseOut": [Function],
      "onMouseOver": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": Object {},
        "arrowContent": null,
        "children": <Markdown
          autoLoadReferences={false}
          className=""
          text="pizza yolo"
        />,
        "destroyTooltipOnHide": false,
        "mouseEnterDelay": 0.3,
        "mouseLeaveDelay": 0.1,
        "overlay": "1",
        "placement": "bottom",
        "prefixCls": "rc-tooltip",
        "trigger": Array [
          "hover",
        ],
        "visible": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "autoLoadReferences": false,
          "className": "",
          "text": "pizza yolo",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Tooltip
          align={Object {}}
          arrowContent={null}
          destroyTooltipOnHide={false}
          mouseEnterDelay={0.3}
          mouseLeaveDelay={0.1}
          overlay="1"
          placement="bottom"
          prefixCls="rc-tooltip"
          trigger={
            Array [
              "hover",
            ]
          }
          visible={false}
        >
          <Markdown
            autoLoadReferences={false}
            className=""
            text="pizza yolo"
          />
        </Tooltip>,
        "className": "TodayReference",
        "onBlur": [Function],
        "onFocus": [Function],
        "onMouseOut": [Function],
        "onMouseOver": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": Object {},
          "arrowContent": null,
          "children": <Markdown
            autoLoadReferences={false}
            className=""
            text="pizza yolo"
          />,
          "destroyTooltipOnHide": false,
          "mouseEnterDelay": 0.3,
          "mouseLeaveDelay": 0.1,
          "overlay": "1",
          "placement": "bottom",
          "prefixCls": "rc-tooltip",
          "trigger": Array [
            "hover",
          ],
          "visible": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "autoLoadReferences": false,
            "className": "",
            "text": "pizza yolo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`render reference without auto load - loading 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodayReference
    autoLoad={false}
    fetchLog={[MockFunction]}
    uuid="1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Tooltip
        align={Object {}}
        arrowContent={null}
        destroyTooltipOnHide={false}
        mouseEnterDelay={0.3}
        mouseLeaveDelay={0.1}
        overlay="1"
        placement="bottom"
        prefixCls="rc-tooltip"
        trigger={
          Array [
            "hover",
          ]
        }
        visible={false}
      >
        <div
          className="TodayReference__Loading text-muted"
        >
          <em>
            <i
              className="fas fa-circle-notch fa-spin TodayReference__Icon"
            />
            Loading 
            <code>
              1
            </code>
            ...
          </em>
        </div>
      </Tooltip>,
      "className": "TodayReference",
      "onBlur": [Function],
      "onFocus": [Function],
      "onMouseOut": [Function],
      "onMouseOver": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": Object {},
        "arrowContent": null,
        "children": <div
          className="TodayReference__Loading text-muted"
        >
          <em>
            <i
              className="fas fa-circle-notch fa-spin TodayReference__Icon"
            />
            Loading 
            <code>
              1
            </code>
            ...
          </em>
        </div>,
        "destroyTooltipOnHide": false,
        "mouseEnterDelay": 0.3,
        "mouseLeaveDelay": 0.1,
        "overlay": "1",
        "placement": "bottom",
        "prefixCls": "rc-tooltip",
        "trigger": Array [
          "hover",
        ],
        "visible": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <em>
            <i
              className="fas fa-circle-notch fa-spin TodayReference__Icon"
            />
            Loading 
            <code>
              1
            </code>
            ...
          </em>,
          "className": "TodayReference__Loading text-muted",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fas fa-circle-notch fa-spin TodayReference__Icon"
              />,
              "Loading ",
              <code>
                1
              </code>,
              "...",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-circle-notch fa-spin TodayReference__Icon",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "Loading ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1",
              },
              "ref": null,
              "rendered": "1",
              "type": "code",
            },
            "...",
          ],
          "type": "em",
        },
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Tooltip
          align={Object {}}
          arrowContent={null}
          destroyTooltipOnHide={false}
          mouseEnterDelay={0.3}
          mouseLeaveDelay={0.1}
          overlay="1"
          placement="bottom"
          prefixCls="rc-tooltip"
          trigger={
            Array [
              "hover",
            ]
          }
          visible={false}
        >
          <div
            className="TodayReference__Loading text-muted"
          >
            <em>
              <i
                className="fas fa-circle-notch fa-spin TodayReference__Icon"
              />
              Loading 
              <code>
                1
              </code>
              ...
            </em>
          </div>
        </Tooltip>,
        "className": "TodayReference",
        "onBlur": [Function],
        "onFocus": [Function],
        "onMouseOut": [Function],
        "onMouseOver": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": Object {},
          "arrowContent": null,
          "children": <div
            className="TodayReference__Loading text-muted"
          >
            <em>
              <i
                className="fas fa-circle-notch fa-spin TodayReference__Icon"
              />
              Loading 
              <code>
                1
              </code>
              ...
            </em>
          </div>,
          "destroyTooltipOnHide": false,
          "mouseEnterDelay": 0.3,
          "mouseLeaveDelay": 0.1,
          "overlay": "1",
          "placement": "bottom",
          "prefixCls": "rc-tooltip",
          "trigger": Array [
            "hover",
          ],
          "visible": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <em>
              <i
                className="fas fa-circle-notch fa-spin TodayReference__Icon"
              />
              Loading 
              <code>
                1
              </code>
              ...
            </em>,
            "className": "TodayReference__Loading text-muted",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-circle-notch fa-spin TodayReference__Icon"
                />,
                "Loading ",
                <code>
                  1
                </code>,
                "...",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-circle-notch fa-spin TodayReference__Icon",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "Loading ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1",
                },
                "ref": null,
                "rendered": "1",
                "type": "code",
              },
              "...",
            ],
            "type": "em",
          },
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`render reference without auto load 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodayReference
    autoLoad={false}
    fetchLog={[MockFunction]}
    uuid="1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Tooltip
        align={Object {}}
        arrowContent={null}
        destroyTooltipOnHide={false}
        mouseEnterDelay={0.3}
        mouseLeaveDelay={0.1}
        overlay="1"
        placement="bottom"
        prefixCls="rc-tooltip"
        trigger={
          Array [
            "hover",
          ]
        }
        visible={false}
      >
        <div
          className="TodayReference__Load text-muted"
        >
          <button
            className="btn btn-outline-primary"
            onClick={[Function]}
          >
            Load 
            <code>
              1
            </code>
          </button>
        </div>
      </Tooltip>,
      "className": "TodayReference",
      "onBlur": [Function],
      "onFocus": [Function],
      "onMouseOut": [Function],
      "onMouseOver": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": Object {},
        "arrowContent": null,
        "children": <div
          className="TodayReference__Load text-muted"
        >
          <button
            className="btn btn-outline-primary"
            onClick={[Function]}
          >
            Load 
            <code>
              1
            </code>
          </button>
        </div>,
        "destroyTooltipOnHide": false,
        "mouseEnterDelay": 0.3,
        "mouseLeaveDelay": 0.1,
        "overlay": "1",
        "placement": "bottom",
        "prefixCls": "rc-tooltip",
        "trigger": Array [
          "hover",
        ],
        "visible": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn btn-outline-primary"
            onClick={[Function]}
          >
            Load 
            <code>
              1
            </code>
          </button>,
          "className": "TodayReference__Load text-muted",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Load ",
              <code>
                1
              </code>,
            ],
            "className": "btn btn-outline-primary",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Load ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1",
              },
              "ref": null,
              "rendered": "1",
              "type": "code",
            },
          ],
          "type": "button",
        },
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Tooltip
          align={Object {}}
          arrowContent={null}
          destroyTooltipOnHide={false}
          mouseEnterDelay={0.3}
          mouseLeaveDelay={0.1}
          overlay="1"
          placement="bottom"
          prefixCls="rc-tooltip"
          trigger={
            Array [
              "hover",
            ]
          }
          visible={false}
        >
          <div
            className="TodayReference__Load text-muted"
          >
            <button
              className="btn btn-outline-primary"
              onClick={[Function]}
            >
              Load 
              <code>
                1
              </code>
            </button>
          </div>
        </Tooltip>,
        "className": "TodayReference",
        "onBlur": [Function],
        "onFocus": [Function],
        "onMouseOut": [Function],
        "onMouseOver": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": Object {},
          "arrowContent": null,
          "children": <div
            className="TodayReference__Load text-muted"
          >
            <button
              className="btn btn-outline-primary"
              onClick={[Function]}
            >
              Load 
              <code>
                1
              </code>
            </button>
          </div>,
          "destroyTooltipOnHide": false,
          "mouseEnterDelay": 0.3,
          "mouseLeaveDelay": 0.1,
          "overlay": "1",
          "placement": "bottom",
          "prefixCls": "rc-tooltip",
          "trigger": Array [
            "hover",
          ],
          "visible": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-outline-primary"
              onClick={[Function]}
            >
              Load 
              <code>
                1
              </code>
            </button>,
            "className": "TodayReference__Load text-muted",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Load ",
                <code>
                  1
                </code>,
              ],
              "className": "btn btn-outline-primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Load ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1",
                },
                "ref": null,
                "rendered": "1",
                "type": "code",
              },
            ],
            "type": "button",
          },
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
